exonum/src/runtime/mod.rs

```diff
@@ -317,38 +326,41 @@ pub trait Runtime: Send + fmt::Debug + 'static {
 
     /// Permanently adds a service to the runtime.
     ///
-    /// This method is called *once* for a specific service instance during `Runtime` lifetime:
+    /// This method is called *once* for a specific service instance during the `Runtime` lifetime:
     ///
     /// - For newly added instances, the method is called when the fork with the corresponding
     ///   `start_adding_service()` call is committed.
     /// - After a node restart, the method is called for all existing service instances.
     ///
-    /// It is guaranteed that `start_adding_service()` was called with the same `spec` earlier
-    /// and returned `Ok(())`. The results of the call (i.e., changes to the blockchain state)
-    /// are guaranteed to be persisted from the call.
+    /// Calling this method guarantees that `start_adding_service()` has been called with the same `spec`
```

Comment:
Maybe replace "calling" with "invocation" to avoid weird "Calling guarantees that ... has been called"?